docs/creating_component_index.md

```diff
@@ -48,7 +48,7 @@ In addition a manifest file is required with below keys as the bare minimum. Cre
 }
 ```
 
-To load this, add `hello_state:` to your `configuration.yaml` file. 
+After creating the file structure, restart Home Assistant to make your new integration available.  Next, add `hello_state:` to your `configuration.yaml` file to enable it, and restart Home Assistant again to load it.  You should see a message in the log indicating that the integration was loaded and a new entity called `hello_state.world`.
```

Comment:
Why should you restart twice? One restart is enough.

Reply:
If you restart from within HA, you cannot add `hello_state:` to configuration.yaml before restarting once with your custom component in place. If you do it all at once, the restart will fail the implicit config check and refuse to restart. This is a common stumbling point for anyone trying custom components that don't use config flow, and isn't common knowledge now that so much has moved to config flow.

Reply:
Then I think we should clarify when two restarts are required and why. If using a development environment, normally you don't restart Home Assistant from within Home Assistant.

Reply:
At least twice that I can recall, folks dabbling in development using their prod HA installation ran into this on Discord. I imagine that they would switch to a dev environment if/when they start developing for real, but I can see folks dipping their toes in the water first. That page does start with a mention of the dev environment and scaffold, but then goes on to say 'really, you just need to do this...'. That's the path the folks I encountered took.

I'll take another stab at the wording.